src/store/actions.js

```diff
@@ -502,30 +502,19 @@ export default {
 			newUid: uid,
 		})
 	},
-	deleteMessage({getters, commit}, envelope) {
-		const folder = getters.getFolder(envelope.accountId, envelope.folderId)
-		commit('removeEnvelope', {
-			accountId: envelope.accountId,
-			folderId: envelope.folderId,
-			id: envelope.id,
-		})
+	deleteMessage({getters, commit}, accountId, folderId, id) {
+		commit('removeEnvelope', {accountId, folderId, id})
 
-		return deleteMessage(envelope.accountId, envelope.folderId, envelope.id)
+		return deleteMessage(accountId, folderId, id)
 			.then(() => {
-				commit('removeMessage', {
-					accountId: envelope.accountId,
-					folder,
-					id: envelope.id,
-				})
+				const folder = getters.getFolder(accountId, folderId)
+				commit('removeMessage', {accountId, folder, id})
```

Comment:
Wouldn't hurt have a `undefined` check here. Right now we *should* always have the folder. But if we don't have it locally, we can just carry on as well.

Reply:
Oh, right, I keep forgetting that things can just be undefined in JS 😂 On it!

Reply:
I'm not 100% this is what you meant? Also, added some logging which.. double check please 😇